internal/common/common.go

```diff
@@ -162,9 +160,29 @@ func (p *ProvenanceIR) GetRepoURIs() []string {
 	return *p.repoURIs
 }
 
-// FromProvenance validates and converts a provenance of arbitrary type to ProvenanceIR
-// TODO(#165): Remove types.ValidatedProvenance and perform the conversion directly on an intoto.statement.
-func FromProvenance(prov *types.ValidatedProvenance) (*ProvenanceIR, error) {
+// WithBinaryName sets the binary name when creating a new ProvenanceIR.
+func WithBinaryName(repoURIs string) func(p *ProvenanceIR) {
+	return func(p *ProvenanceIR) {
+		p.binaryName = &repoURIs
+	}
+}
+
+// HasBinaryName returns true if the binary name has been set in the ProvenanceIR.
+func (p *ProvenanceIR) HasBinaryName() bool {
+	return p.binaryName != nil
+}
+
+// GetBinaryName returns the binary name. Returns an error if the binary name has not been set.
+func (p *ProvenanceIR) GetBinaryName() (string, error) {
```

Comment:
All provenances (including SLSAv02) should have a BinaryName. So I think this function can be simplified to return just a string.